### PR TITLE
Feature/wa 393 creation

### PR DESCRIPTION
**Pull request by DmitryBespalov**

### CODE REVIEW

File: MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift [hunk 1]
```diff
@@ -89,7 +89,11 @@ public class WalletApplicationService: Assertable {
     }
 
     public var selectedWalletAddress: String? {
-        return (try? findSelectedWallet())?.address?.value
+        do {
```
Comment (sche) [inline MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift:92]: you can leave as it was before

File: MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift [hunk 2]
```diff
@@ -197,23 +203,26 @@ public class WalletApplicationService: Assertable {
...
 
-    private func didFinishDeployment(success: Bool, error: Swift.Error?) {
+    private func didFinishDeployment(success: Bool) {
         if success {
             do {
                 try fetchBalance()
                 try markDeploymentSuccess()
-                try finishDeployment()
                 try removePaperWallet()
+                try finishDeployment()
```
Comment (sche) [inline MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift:219]: so if we cancel at this stage there should not be a configured paper wallet, right?

File: MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift [hunk 3]
```diff
@@ -197,23 +203,26 @@ public class WalletApplicationService: Assertable {
...
             } catch let error {
                 ApplicationServiceRegistry.logger.fatal("Failed to save success deployment state", error: error)
                 try? markDeploymentFailed()
             }
         } else {
-            ApplicationServiceRegistry.logger.fatal("Failed to deploy wallet in blockchain", error: error!)
+            ApplicationServiceRegistry.logger.fatal("Blockchain transaction failed")
```
Comment (sche) [inline MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift:225]: we will not be notified about it, right?

File: MultisigWallet/MultisigWalletDomainModel/Wallet.swift [hunk 4]
```diff
@@ -151,7 +151,7 @@ public class Wallet: IdentifiableEntity<WalletID> {
     }
 
     public func markDeploymentFailed() throws {
-        try assert(status: .deploymentAcceptedByBlockchain)
+        try assert(statusIsOneOf: .deploymentAcceptedByBlockchain, .addressKnown)
```
Comment (sche) [inline MultisigWallet/MultisigWalletDomainModel/Wallet.swift:155]: please update class doc

File: SafeUIKit/SafeAppUI/OnboardingFlow/SetupSafeFlow/NewSafeFlow/AbortSafeCreationAlertController.swift [hunk 5]
```diff
@@ -22,8 +22,8 @@ class AbortSafeCreationAlertController: SafeAlertController {
         static let message = LocalizedString("pending_safe.abort_alert.message", comment: "Message body of abort alert")
         static let abortTitle = LocalizedString("pending_safe.abort_alert.abort",
                                                 comment: "Abort safe creation button title")
-        static let continueTitle = LocalizedString("pending_safe.abort_alert.continue",
-                                                   comment: "Continue safe creation button title")
+        static let cancelTitle = LocalizedString("pending_safe.abort_alert.cancel",
+                                                 comment: "Continue safe creation button title")
```
Comment (sche) [inline SafeUIKit/SafeAppUI/OnboardingFlow/SetupSafeFlow/NewSafeFlow/AbortSafeCreationAlertController.swift]: please update comment

File: MultisigWallet/MultisigWalletDomainModel/Wallet.swift (hunk 4 above)
Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletDomainModel/Wallet.swift:155]: ok

File: SafeUIKit/SafeAppUI/OnboardingFlow/SetupSafeFlow/NewSafeFlow/AbortSafeCreationAlertController.swift (hunk 5 above)
Comment (DmitryBespalov) [reply SafeUIKit/SafeAppUI/OnboardingFlow/SetupSafeFlow/NewSafeFlow/AbortSafeCreationAlertController.swift]: ok

File: MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift (hunk 3 above)
Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift:225]: no, now it is only local log

File: MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift (hunk 2 above)
Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift:219]: yes

File: MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift (hunk 1 above)
Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift:92]: i like it more here

File: MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift (hunk 3 above)
Comment (sche) [reply MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift:225]: but any fatal should mean that we should be aware of it

File: MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift [hunk 6]
```diff
@@ -197,23 +203,26 @@ public class WalletApplicationService: Assertable {
...
         try markDeploymentAcceptedByBlockchain()
+        let isSucdess = try service.waitForPendingTransaction(hash: hash)
+        didFinishDeployment(success: isSucdess)
     }
 
-    private func didFinishDeployment(success: Bool, error: Swift.Error?) {
+    private func didFinishDeployment(success: Bool) {
         if success {
             do {
                 try fetchBalance()
```
Comment (sche) [inline MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift:216]: lets move it to MainViewController or at least make this optional try

Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift:216]: I removed this in the upcoming pr.

File: MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift (hunk 3 above)
Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift:225]: yes, this is coming later.

**Pull request merged**